step-generation/src/utils/pythonFileUtils.ts

```diff
@@ -250,14 +253,29 @@ export function getLoadPipettes(
       const pipetteName = isFlexPipette(name)
         ? getFlexNameConversion(spec)
         : name
-      const tiprackPythonNames = tiprackDefURI
-        .flatMap(defURI =>
-          Object.values(labwareEntities).filter(
-            lw => lw.labwareDefURI === defURI
-          )
-        )
+      const allTipracks = tiprackDefURI.reduce(
+        (acc: LabwareEntity[], defURI) => {
+          for (const lw of Object.values(labwareEntities)) {
+            if (lw.labwareDefURI === defURI) {
+              acc.push(lw)
+            }
+          }
+          return acc
+        },
+        []
+      )
+      //  list off-deck labware last to match getNextTip logic
+      const tiprackPythonNames = allTipracks
```

Comment:
probably protecting the original array would be better.

```suggestion
      const tiprackPythonNames = [...allTipracks]
```